ipc/identity/src/evm/mod.rs

```diff
@@ -69,3 +79,50 @@ pub fn random_key_info() -> KeyInfo {
     let key = ethers::core::k256::SecretKey::random(&mut rand::thread_rng());
     KeyInfo::new(key.to_bytes().to_vec())
 }
+
+#[cfg(feature = "with-ethers")]
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub struct EthKeyAddress {
+    inner: ethers::types::Address,
+}
+
+#[cfg(feature = "with-ethers")]
+impl From<ethers::types::Address> for EthKeyAddress {
+    fn from(inner: ethers::types::Address) -> Self {
+        EthKeyAddress { inner }
+    }
+}
+
+#[cfg(feature = "with-ethers")]
+impl From<EthKeyAddress> for ethers::types::Address {
+    fn from(val: EthKeyAddress) -> Self {
+        val.inner
+    }
+}
+
+#[cfg(feature = "with-ethers")]
+impl Defaultable for EthKeyAddress {
+    fn default_key() -> Self {
+        Self {
+            inner: ethers::types::Address::default(),
+        }
+    }
+}
+
+#[cfg(feature = "with-ethers")]
+impl ToString for EthKeyAddress {
+    fn to_string(&self) -> String {
+        self.inner.to_string()
+    }
+}
+
+#[cfg(feature = "with-ethers")]
```

Comment:
maybe we can put all the `#[cfg(feature = "with-ethers")]` code into a separate module/file, then we just enable `#[cfg(feature = "with-ethers")]` in mod file.

Reply:
I'll probably leave this one for the next refactor if you don't mind so I don't mess up this PR more.